internal/module/module.go

```diff
@@ -26,34 +25,20 @@ type TemplateModule struct {
 	Config                moduleconfig.ModuleConfig
 }
 
-type ProgressTracking struct {
-	sync.Mutex
-	downloaded map[string]int
-}
-
-// Init downloads the remote template files and parses the module config yaml
-func NewTemplateModule(moduleCfg config.ModuleInstance) (*TemplateModule, error) {
-	var templateModule TemplateModule
-	templateModule.Source = moduleCfg.Source
-	templateModule.Params = moduleCfg.Params
-	templateModule.Overwrite = moduleCfg.Overwrite
-	templateModule.Output = moduleCfg.Output
-
-	p := &ProgressTracking{}
-	sourcePath := GetSourceDir(templateModule.Source)
-
-	if !IsLocal(templateModule.Source) {
-		err := getter.Get(sourcePath, templateModule.Source, getter.WithProgress(p))
+// FetchModule downloads the remote module source (or loads the local files) and parses the module config yaml
+func FetchModule(source string) (moduleconfig.ModuleConfig, error) {
```

Comment:
Now accepts a single module source (directory or url) and returns the config of that project.